lib/ui/painting/codec.cc

```diff
@@ -217,10 +217,10 @@ static void InstantiateImageCodec(Dart_NativeArguments args) {
     ImageDecoder::ImageDescriptor descriptor;
     descriptor.decompressed_image_info = image_info;
 
-    if (targetWidth != kDoNotResizeDimension) {
+    if (targetWidth > kDoNotResizeDimension) {
```

Comment:
You can just say `> 0` and remove the constant. This file doesn't need to know or care about `kDoNotResizeDimension`. You are getting away with this being safe because it is -1 now. If it was any other negative number, you could still cause casts of unsigned ints.